xscen/catalog.py

```diff
@@ -118,18 +119,28 @@ def _parse_list_of_strings(elem):
     return (elem,)
 
 
+# def _parse_dates(elem):
+#     try:
+#         if isinstance(elem, str):
+#             return pd.Timestamp(elem).to_period("H")
+#         return pd.DatetimeIndex(elem).to_period("H")
+#     except pd.errors.OutOfBoundsDatetime:
+#         warnings.warn(
+#             "Somes dates are out of the datetime64[ns] range, switching to slower direct period parsing."
+#         )
+#         if isinstance(elem, str):
+#             return pd.Period(elem, freq="H")
+#         return pd.PeriodIndex(elem, freq="H")
 def _parse_dates(elem):
-    try:
-        if isinstance(elem, str):
-            return pd.Timestamp(elem).to_period("H")
-        return pd.DatetimeIndex(elem).to_period("H")
-    except pd.errors.OutOfBoundsDatetime:
-        warnings.warn(
-            "Somes dates are out of the datetime64[ns] range, switching to slower direct period parsing."
-        )
```

Comment:
I'm not sure I see the advantage because now there is no switch, it always go through the same code. Also, `time.isnull()` also picks up `fx` entries, so the `PeriodIndex` call is almost always non-empty, but that doesn't mean that dates are out of the range.

I would prefer document it elsewhere in the doc than raise a warning each time we read a catalog.

Reply:
Alright, good with me!